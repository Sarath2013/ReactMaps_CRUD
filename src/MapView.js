import React from 'react';
import PropTypes from 'prop-types'
import { Map, GoogleApiWrapper, Marker } from 'google-maps-react';

const mapStyles = {
  width: '100%',
  height: '100%'
};

const MapComponent = (props) => {
  function loadMarkers() {
    return props.locations.map((location, index) => {
      return <Marker key={index} id={index} position={{
        lat: location.lat,
        lng: location.lng
      }}
        onClick={updateLocations} />
    })
  }

  function updateLocations() {
    props.update(props.locations);
  }

  return (
    <Map
      google={props.google}
      className='map-container'
      style={mapStyles}
      initialCenters={{ lat: 52.520008, lng: 13.404954 }}
      center={props.activeMarkerValues}
    >
      {loadMarkers()}
    </Map>
  );
}

MapComponent.propTypes = {
  locations: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    lat: PropTypes.number.isRequired,
    lng: PropTypes.number.isRequired
  })).isRequired,
  activeMarkerValues: PropTypes.shape({
    lat: PropTypes.number.isRequired,
    lng: PropTypes.number.isRequired
  }).isRequired,
  update: PropTypes.func.isRequired
}

export default GoogleApiWrapper({
  apiKey: process.env.REACT_APP_MAPS_API_KEY
})(MapComponent);
