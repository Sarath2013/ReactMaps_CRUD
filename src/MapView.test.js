import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MapView from './MapView';

const props = {
  locations: [{
    name: 'Berlin',
    lat: 52.52,
    lng: 13.405
  },
  {
    name: 'Munich',
    lat: 48.1351,
    lng: 11.582
  }],
  update: jest.fn(),
  activeMarkerValues: {
    lat: 52.52,
    lng: 13.405
  }
};
let wrapper;

describe("Rendering MapView", () => {
  beforeAll(() => {
  });

  afterAll(() => {
    wrapper = null;
  });

  test('renders snapshot', () => {
    wrapper = shallow(<MapView {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.instance().props.locations.length).toBe(2);
    expect(wrapper.instance().props.activeMarkerValues).toEqual({
      lat: 52.52,
      lng: 13.405
    });
  });
});

