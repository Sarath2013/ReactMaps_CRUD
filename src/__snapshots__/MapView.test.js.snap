// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering MapView renders snapshot 1`] = `<DefaultLoadingContainer />`;
